synapse/config/oidc_config.py

```diff
@@ -280,8 +280,8 @@ def _parse_oidc_provider_configs(config: JsonDict) -> Iterable["OidcProviderConf
     """
     validate_config(MAIN_CONFIG_SCHEMA, config, ())
 
-    for p in config.get("oidc_providers") or []:
-        yield _parse_oidc_config_dict(p)
+    for i, p in enumerate(config.get("oidc_providers")) or []:
```

Comment:
I suspect this should be:
```suggestion
    for i, p in enumerate(config.get("oidc_providers") or []):
```

This should handle bogus values better (e.g. `oidc_providers: null` or something.

Reply:
:facepalm: thanks